src/qt/walletmodel.cpp

```diff
@@ -36,10 +40,20 @@
 #include <QSet>
 #include <QTimer>
 
+#include <boost/foreach.hpp>
+#include <map>
+#include <univalue.h>
 
-WalletModel::WalletModel(const PlatformStyle *platformStyle, CWallet *_wallet, OptionsModel *_optionsModel, QObject *parent) :
-    QObject(parent), wallet(_wallet), optionsModel(_optionsModel), addressTableModel(0),
+/* We need the following functions declared for use in the QT UI */
+extern UniValue name_new(const JSONRPCRequest& request);
+extern UniValue name_update(const JSONRPCRequest& request);
+extern UniValue name_firstupdate(const JSONRPCRequest& request);
+extern UniValue gettransaction(const JSONRPCRequest& request);
+
+WalletModel::WalletModel(const PlatformStyle *platformStyle, CWallet *_wallet, OptionsModel *_optionsModel, QWidget *parent) :
+    QWidget(parent), wallet(_wallet), optionsModel(_optionsModel), addressTableModel(0),
```

Comment:
Why do you change parent from `QObject` to `QWidget`, is that related to your change?

Reply:
In order to get access to certain Qt utils (like `QMessageBox`, certain buttons, etc) we need a `QWidget`, not a `QObject` as an inherited class. In order to use a `QWidget`, you need a parent which is also declared as a `QWidget`. If you'll notice, most of the other dialogs and pages inherit `QWidget` and also require a parent of `QWidget`.

Reply:
Sounds good, thanks for the explanation!